lib/ecto/changeset.ex

```diff
@@ -1072,7 +1072,7 @@ defmodule Ecto.Changeset do
       apply_changes(changeset)
 
   """
-  @spec apply_changes(t) :: Ecto.Schema.t
+  @spec apply_changes(t) :: Ecto.Schema.t | data
```

Comment:
I see. I was confused by the `data` type not fully representing the types of the `data` field.